File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i = 0
        j = len(nums)-1
        while i < j:
            if nums[i] + nums[j] == target:
                return [i, j]
            elif j == i+1:
                j = len(nums)-1
                i += 1
            else:
                j -= 1

        

File: Data Structures & Algorithms/binary-search/submission-0.go
func search(nums []int, target int) int {
    j := (len(nums)-1)
    i := 0

    for i <= j{
        
        mid := i + (j - i)/2

        if nums[mid] == target {
            return mid
        } else if nums[mid] > target {
            j = mid - 1
        } else {
            i = mid + 1
        }
        
    }

    return -1
}
